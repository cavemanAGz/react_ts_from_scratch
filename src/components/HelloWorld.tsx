import React from "react"

const HelloWorld: React.FC = () => {
    return (
        <>
            <h1>Hello World</h1>
            <hr />
            <h2>Hello Caveman!</h2>
            <p>
                process.env.PRODUCTION: <b>{process.env.PRODUCTION.toString()}</b>
            </p>
            <p>
                process.env.NAME: <b>{process.env.NAME}</b>
            </p>
            <p>
                process.env.VERSION: <b>{process.env.VERSION}</b>
            </p>
            <p>
                process.env.REACT_APP_VERSION:
                <b>{process.env.COMMIT_HASH != null ? process.env.COMMIT_HASH : "IDK???"}</b>
            </p>
        </>
    )
}

export default HelloWorld
