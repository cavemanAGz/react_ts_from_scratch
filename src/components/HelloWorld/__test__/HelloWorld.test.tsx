import React from "react"
import {shallow} from "enzyme"
import HelloWorld from "../HelloWorld"
// import * as Enzyme from "enzyme"
// import Adapter from "@wojtekmaj/enzyme-adapter-react-17"
//
// Enzyme.configure({adapter: new Adapter()})
describe("Testing the Hello World Component", () => {
    test("add 1 + 3 to equal 4", () => {
        expect(4).toBe(4)
    })
    it("Renders Hellos Word Component", () => {
        process.env.PRODUCTION = "TEST"
        process.env.NAME = "React From Scratch *** TEST ***"
        process.env.VERSION = "1.0.0"
        process.env.COMMIT_HASH = "c645248s525344"
        const helloWorld = shallow(<HelloWorld />)
        expect(helloWorld).toMatchSnapshot()
    })
})
