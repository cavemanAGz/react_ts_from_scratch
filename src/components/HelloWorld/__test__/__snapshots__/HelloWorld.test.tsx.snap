// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Hello World Component Renders Hellos Word Component 1`] = `ShallowWrapper {}`;
